coderd/telemetry/telemetry.go

```diff
@@ -0,0 +1,585 @@
+				JobID:               schema.JobID,
+				Name:                schema.Name,
+				ValidationCondition: schema.ValidationCondition,
+			})
+		}
+		return nil
+	})
+	eg.Go(func() error {
+		jobs, err := r.options.Database.GetProvisionerJobsCreatedAfter(ctx, createdAfter)
+		if err != nil {
+			return xerrors.Errorf("get provisioner jobs: %w", err)
+		}
+		snapshot.ProvisionerJobs = make([]ProvisionerJob, 0, len(jobs))
+		for _, job := range jobs {
+			snapJob := ProvisionerJob{
+				ID:             job.ID,
+				OrganizationID: job.OrganizationID,
+				InitiatorID:    job.InitiatorID,
+				CreatedAt:      job.CreatedAt,
+				UpdatedAt:      job.UpdatedAt,
+				Error:          job.Error.String,
+				Type:           job.Type,
+			}
+			if job.StartedAt.Valid {
+				snapJob.StartedAt = &job.StartedAt.Time
+			}
+			if job.CanceledAt.Valid {
+				snapJob.CanceledAt = &job.CanceledAt.Time
+			}
+			if job.CompletedAt.Valid {
+				snapJob.CompletedAt = &job.CompletedAt.Time
+			}
+			snapshot.ProvisionerJobs = append(snapshot.ProvisionerJobs, snapJob)
+		}
+		return nil
+	})
+	eg.Go(func() error {
+		templates, err := r.options.Database.GetTemplates(ctx)
+		if err != nil {
+			return xerrors.Errorf("get templates: %w", err)
+		}
+		snapshot.Templates = make([]Template, 0, len(templates))
+		for _, dbTemplate := range templates {
+			snapshot.Templates = append(snapshot.Templates, Template{
+				ID:              dbTemplate.ID,
+				CreatedBy:       dbTemplate.CreatedBy.UUID,
+				CreatedAt:       dbTemplate.CreatedAt,
+				UpdatedAt:       dbTemplate.UpdatedAt,
+				OrganizationID:  dbTemplate.OrganizationID,
+				Deleted:         dbTemplate.Deleted,
+				ActiveVersionID: dbTemplate.ActiveVersionID,
+				Name:            dbTemplate.Name,
+				Description:     dbTemplate.Description != "",
+			})
+		}
+		return nil
+	})
+	eg.Go(func() error {
+		templateVersions, err := r.options.Database.GetTemplateVersionsCreatedAfter(ctx, createdAfter)
+		if err != nil {
+			return xerrors.Errorf("get template versions: %w", err)
+		}
+		snapshot.TemplateVersions = make([]TemplateVersion, 0, len(templateVersions))
+		for _, version := range templateVersions {
+			snapVersion := TemplateVersion{
+				ID:             version.ID,
+				CreatedAt:      version.CreatedAt,
+				OrganizationID: version.OrganizationID,
+				JobID:          version.JobID,
+			}
+			if version.TemplateID.Valid {
+				snapVersion.TemplateID = &version.TemplateID.UUID
+			}
+			snapshot.TemplateVersions = append(snapshot.TemplateVersions, snapVersion)
+		}
+		return nil
+	})
+	eg.Go(func() error {
+		users, err := r.options.Database.GetUsers(ctx, database.GetUsersParams{})
+		if err != nil {
+			return xerrors.Errorf("get users: %w", err)
+		}
+		snapshot.Users = make([]User, 0, len(users))
+		for _, dbUser := range users {
+			emailHashed := ""
+			atSymbol := strings.LastIndex(dbUser.Email, "@")
+			if atSymbol >= 0 {
+				hash := sha256.Sum256([]byte(dbUser.Email[:atSymbol]))
```

Comment:
This hash seems easily crackable tbh. What if we just used `{user_id}@domain.com`?

Reply:
I want it to be traceable across deployments. We could use a greater hash, but I agree it's still crackable regardless.

Reply:
Ah yeah, I forgot about that. This seems reasonable then.

Reply:
I'll add a comment here for that